packages/react-router-dom/modules/BrowserPrompt.js

```diff
@@ -0,0 +1,80 @@
+import React, { PropTypes } from 'react'
+import { Prompt } from 'react-router'
+
+const BEFOREUNLOAD = "beforeunload"; // see: http://stackoverflow.com/questions/39094138/reactjs-event-listener-beforeunload-added-but-not-removed
+
+var UnloadComponent = ComposedComponent => class extends React.Component {
+  static contextTypes = {
+    router: PropTypes.shape({
+        history: PropTypes.shape({
+            block: PropTypes.func.isRequired
+        }).isRequired
+    }).isRequired
+  }
+
+  static propTypes = {
+    when: PropTypes.bool,
+    message: PropTypes.oneOfType([
+        PropTypes.func,
+        PropTypes.string
+    ]).isRequired,
+    beforeUnload: PropTypes.oneOfType([
+        PropTypes.func,
+        PropTypes.bool
```

Comment:
I didn't want to lock the developer into an all or nothing.

For example, we might want to do the following.
- Stop navigating to different page and stop unload
- Allow navigation to different page but stop unload
- Stop navigation but conditionally allow unload 

I can see people using this as a general (all pages) onbeforeunload. If this is tied into the Prompt then prop, they would be unable to move to different pages.